src/React.rei

```diff
@@ -565,15 +566,81 @@ module Uncurried: {
 };
 
 [@mel.module "react"]
-external startTransition: ([@mel.uncurry] (unit => unit)) => unit = "startTransition";
+external startTransition: ([@mel.uncurry] (unit => unit)) => unit =
+  "startTransition";
 
 [@mel.module "react"]
 external useTransition: unit => (bool, callback(callback(unit, unit), unit)) =
   "useTransition";
 
 module Experimental: {
   /* This module is used to bind to APIs for future versions of React. There is no guarantee of backwards compatibility or stability. */
-  [@mel.module "react"] external use: Js.Promise.t('a) => 'a = "use";
+  [@mel.module "react"] external usePromise: Js.Promise.t('a) => 'a = "use";
+  [@mel.module "react"] external useContext: Context.t('a) => 'a = "use";
+
+  [@mel.module "react"]
+  external useOptimistic:
+    ('state, ('state, 'optimisticValue) => 'state) =>
+    ('state, 'optimisticValue => unit) =
+    "useOptimistic";
+
+  [@mel.module "react"]
+  external useTransitionAsync:
+    unit => (bool, callbackAsync(callbackAsync(unit, unit), unit)) =
+    "useTransition";
```

Comment:
Prob out of scope, but now that new bindings are being added, and given they're new and experimental, it's good chance to use chatgpt to add comments to them? i got this for this one with some tweaking:

```suggestion
  /**
   * `useTransitionAsync` is a binding to React's `useTransition` hook.
   * 
   * This hook lets you update state without blocking the UI.
   * 
   * ### Return Value
   * - `(bool, callbackAsync(callbackAsync(unit, unit), unit))`: A tuple containing:
   *   - `bool`: Whether the transition is pending.
   *   - A function to start the transition.
   * 
   * ### Example
   * ```reason
   * let (isPending, startTransition) = useTransitionAsync();
   *
   * /* Start a transition */
   * startTransition(() => {
   *   /* Deferred update logic */
   * });
   * 
   * if (isPending) {
   *   /* Show loading indicator */
   * }
   * ```
   */
  [@mel.module "react"]
  external useTransitionAsync:
    unit => (bool, callbackAsync(callbackAsync(unit, unit), unit)) =
    "useTransition";
```

Reply:
I think this is a good idea but I'd recommend doing it in a future PR. this is already quite a bit dense for revewing.